internal/resource/configmap.go

```diff
@@ -96,9 +96,14 @@ func (builder *ServerConfigMapBuilder) Update(object runtime.Object) error {
 		}
 	}
 
-	if builder.Instance.Spec.Rabbitmq.AdvancedConfig != "" {
+	if builder.Instance.Spec.Rabbitmq.AdvancedConfig == "" {
+		delete(configMap.Data, "advanced.config")
```

Comment:
Ooh I didn't know delete was a no-op if the key doesn't exist! 

Given this is quite a clean pattern for advanced now, would it make sense to do the same in envConfig? That way when you remove the config from the CR spec, both of the config files are removed too (rather than having an empty `rabbitmq-env.conf`)?